File: src/main/java/io/github/oxnz/Ingrid/dts/TxResult.java
package io.github.oxnz.Ingrid.dts;

import io.github.oxnz.Ingrid.dts.data.TxRecord;
import org.springframework.core.style.ToStringCreator;

import javax.persistence.*;

@Entity
public class TxResult {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @ManyToOne(cascade = CascadeType.DETACH)
    @JoinColumn(name = "record_id")
    private TxRecord record;
    public boolean succ;
    public String msg;

    public TxResult() {
    }

    public TxResult(TxRecord record, boolean succ, String msg) {
        this.record = record;
        this.succ = succ;
        this.msg = msg;
    }

    @Override
    public String toString() {
        return new ToStringCreator(this)
                .append("id", id)
                .append("record", record)
                .append("succeeded", succ)
                .append("message", msg)
                .toString();
    }
}

File: src/main/scala/io/github/oxnz/Ingrid/tx/TxResult.scala
package io.github.oxnz.Ingrid.tx

import javax.persistence.{Entity, _}

import scala.beans.BeanProperty


@Entity class TxResult(_record: TxRecord, @BeanProperty var succ: Boolean, @BeanProperty var msg: String) {
  @Id
  @BeanProperty
  @GeneratedValue(strategy = GenerationType.AUTO) val id = 0L
  @BeanProperty
  @ManyToOne(cascade = Array(CascadeType.DETACH))
  @JoinColumn(name = "record_id") var record: TxRecord = _record

  def this() {
    this(null, false, null)
  }


  override def toString = s"TxResult(id=$id, record=$record, succ=$succ, msg=$msg)"
}
